Libs/MRML/Core/vtkMRMLBitStreamVolumeNode.cxx

```diff
@@ -0,0 +1,235 @@
+#include "vtkMRMLBitStreamVolumeNode.h"
+
+// OpenIGTLink includes
+//#include "igtlioVideoDevice.h"
+
+// MRML includes
+#include "vtkMRMLScene.h"
+#include "vtkMRMLNRRDStorageNode.h"
+
+// VTK includes
+#include <vtkCollection.h>
+#include <vtkMatrix4x4.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkXMLUtilities.h>
+
+//----------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLBitStreamVolumeNode);
+
+//----------------------------------------------------------------------------
+// vtkMRMLBitStreamVolumeNode methods
+
+//-----------------------------------------------------------------------------
+vtkMRMLBitStreamVolumeNode::vtkMRMLBitStreamVolumeNode()
+{
+  this->compressionDevice = NULL;
+  IsCopied = false;
+  KeyFrameReceived = false;
+  KeyFrameDecoded = false;
+  KeyFrameUpdated = false;
+  MessageBufferValid = false;
+  KeyFrameBuffer = "";
+  MessageBuffer = "";
+}
+
+//-----------------------------------------------------------------------------
+vtkMRMLBitStreamVolumeNode::~vtkMRMLBitStreamVolumeNode()
+{
+  if (compressionDevice != NULL)
+     compressionDevice->Delete();
+}
+
+//---------------------------------------------------------------------------
+int vtkMRMLBitStreamVolumeNode::ObserveOutsideCompressionDevice(vtkMRMLCompressionDeviceNode* device)
+{
+  if (device)
+  {
+    device->AddObserver(device->GetDeviceContentModifiedEvent(), this, &vtkMRMLBitStreamVolumeNode::ProcessDeviceModifiedEvents);
+    //------
+    std::string bitStream = std::string(device->GetContent()->frameMessage);
+    this->SetMessageStream(bitStream);
+    this->SetAndObserveImageData(device->GetContent()->image);
+    this->compressionDevice = device; // should the interal video device point to the external video device?
+    //-------
+    return 1;
+  }
+  return 0;
+}
+
+void vtkMRMLBitStreamVolumeNode::SetMessageStream(std::string buffer)
+{
+  this->MessageBuffer.resize(buffer.length());
+  char * array = new char[buffer.length()];
+  memcpy(array, buffer.c_str(), buffer.length());
+  this->MessageBuffer.assign(array, buffer.length());
+  this->MessageBufferValid = true;
+};
+
+std::string vtkMRMLBitStreamVolumeNode::GetMessageStream()
```

Comment:
Method should probably not include the word `Stream` as the return value is a string